sdks/python/apache_beam/runners/worker/data_plane.py

```diff
@@ -218,7 +249,7 @@ class DataChannel(with_metaclass(abc.ABCMeta, object)):  # type: ignore[misc]
   @abc.abstractmethod
   def input_elements(self,
                      instruction_id,  # type: str
-                     expected_inputs,  # type: Collection[str]
+                     expected_inputs,  # type: Sized
```

Comment:
we just call `len(expected_inputs)`, so we just need to know this implements `__len__`

Reply:
Don't we call `__contains__` as well? I'd rather keep it more fully typed.

Reply:
we only call `__len__`, but I'm happy to leave it as a collection.  updated.